src/main/java/com/singlestore/jdbc/DatabaseMetaData.java

```diff
@@ -171,9 +172,13 @@ private Version getMySQLVersion() throws java.sql.SQLException {
   private ResultSet executeQuery(String sql) throws SQLException {
     Statement stmt =
         connection.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
-    CompleteResult rs = (CompleteResult) stmt.executeQuery(sql);
-    rs.setStatement(null); // bypass Hibernate statement tracking (CONJ-49)
-    rs.useAliasAsName();
```

Comment:
How about just reset this statement to **stmt.setFetchSize(0)** to avoid streaming for metadata?

Reply:
Good idea, I have changed to do it that way instead.